samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -196,88 +159,39 @@ public TestRunner addOverrideConfig(String key, String value) {
   }
 
   /**
-   * Configures {@code stream} with the TestRunner, adds all the stream specific configs to global job configs.
-   * <p>
-   * Every stream belongs to a System (here a {@link CollectionStreamSystemSpec}), this utility also registers the system with
-   * {@link TestRunner} if not registered already. Then it creates and initializes the stream partitions with messages for
-   * the registered System
-   * <p>
-   * @param stream represents the stream that is supposed to be configured with {@link TestRunner}
-   * @return calling instance of {@link TestRunner} with {@code stream} configured with it
+   * Configures an inmemory input stream with the TestRunner, adds all the stream and system specific configs
+   * to global job configs.
+   * @param descriptor describes the stream that is supposed to be input to Samza application
+   * @return calling instance of {@link TestRunner} with input stream configured with it
    */
-  public TestRunner addInputStream(CollectionStream stream) {
-    Preconditions.checkNotNull(stream);
-    registerSystem(stream.getSystemName());
-    initializeInput(stream);
-    stream.setTestId(testId);
+  public TestRunner addInputStream(InMemoryInputDescriptor descriptor) {
+    Preconditions.checkNotNull(descriptor);
+    String systemName = descriptor.getSystemName();
+    String streamName = (String) descriptor.getPhysicalName().orElse(descriptor.getStreamId());
     if (configs.containsKey(TaskConfig.INPUT_STREAMS())) {
       configs.put(TaskConfig.INPUT_STREAMS(),
-          configs.get(TaskConfig.INPUT_STREAMS()).concat("," + stream.getSystemName() + "." + stream.getPhysicalName()));
+          configs.get(TaskConfig.INPUT_STREAMS()).concat("," + systemName + "." + streamName));
     } else {
-      configs.put(TaskConfig.INPUT_STREAMS(), stream.getSystemName() + "." + stream.getPhysicalName());
+      configs.put(TaskConfig.INPUT_STREAMS(), systemName + "." + streamName);
     }
-    stream.getStreamConfig().forEach((key, val) -> {
-        configs.putIfAbsent((String) key, (String) val);
-      });
-
+    addConfigs(descriptor.toConfig());
```

Comment:
Is toConfig needed because the in-memory descriptors aren't actually used in the graph spec? If so, could you please put a comment about that, since that detail is a little hard to figure out.